snet_cli/commands.py

```diff
@@ -96,8 +124,16 @@ def get_eth_endpoint(self):
     def get_wallet_index(self):
         return int(get_field_from_args_or_session(self.config, self.args, "wallet_index"))
 
-    def get_gas_price(self):
-        return int(get_field_from_args_or_session(self.config, self.args, "gas_price"))
+    def get_gas_price_param(self):
+         return get_field_from_args_or_session(self.config, self.args, "gas_price")
+
+    def get_gas_price_verbose(self):
+        # gas price is not given explicitly in Wei
+        if (self.w3.eth.gasPriceStrategy.is_going_to_calculate()):
+            self._printerr("# Calculating gas price. It might take ~60 seconds.")
```

Comment:
I test this like 5x...and the delay to find the gas price is about ~35s (Ropsten).
But even with this delay, the transaction seems to take the same time as before on Ropsten. =/
Is this the expected behavior?

Reply:
Yes. On ropsten you can set gas_price = 1 Wei, and it still will pass with the same speed.

This PR will affect only the mainnet. But it should simplify the life of users significantly, otherwise users will need to manually set the gas-price, but to do so user should understand what the gas price is...